libs/ngworker/lumberjack/src/lib/logging/lumberjack-logger.builder.spec.ts

```diff
@@ -0,0 +1,132 @@
+  let lumberjackService: LumberjackService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [LumberjackModule.forRoot()],
+      providers: [{ provide: LumberjackTimeService, useClass: FakeTimeService }],
+    });
+
+    fakeTime = resolveDependency(LumberjackTimeService) as FakeTimeService;
+    lumberjackService = (resolveDependency(LumberjackService) as unknown) as LumberjackService;
+    spyOn(lumberjackService, 'log').and.stub();
+  });
+
+  describe('Base builder usage', () => {
+    lumberjackLogLevels.forEach((level) =>
+      it(`sends a ${level} level log`, () => {
+        const builder = new LumberjackLoggerBuilder(lumberjackService, fakeTime, level, testMessage);
+        const logFunction = builder.build();
+        logFunction();
+        const expectedLog: LumberjackLog = {
+          message: testMessage,
+          level,
+          createdAt: fakeTime.getUnixEpochTicks(),
+          payload: undefined,
+          scope: '',
+        };
+
+        expect(lumberjackService.log).toHaveBeenCalledWith(expectedLog);
+      })
+    );
+  });
+
+  it('logs the specified scope', () => {
+    const level = LumberjackLevel.Critical;
+    const scope = 'Test Scope';
+    const builder = new LumberjackLoggerBuilder(lumberjackService, fakeTime, level, testMessage);
+    const logFunction = builder.withScope(scope).build();
+    logFunction();
+    const expectedLog: LumberjackLog = {
+      message: testMessage,
+      level,
+      createdAt: fakeTime.getUnixEpochTicks(),
+      payload: undefined,
+      scope,
+    };
+
+    expect(lumberjackService.log).toHaveBeenCalledWith(expectedLog);
+  });
+
+  describe('LumberjackLogPayload', () => {
+    const level = LumberjackLevel.Critical;
+    const scope = 'Test Scope';
+    const payload: TestPayload = {
+      testProperty: 'TEST_PROPERTY',
+    };
+    let builder: LumberjackLoggerBuilder<TestPayload>;
+    beforeEach(() => {
+      builder = new LumberjackLoggerBuilder<TestPayload>(
+        (lumberjackService as unknown) as LumberjackService<TestPayload>,
+        fakeTime,
+        level,
+        testMessage
+      );
+    });
+
+    it('logs the specified payload', () => {
+      const logFunction = builder.withScope(scope).build();
+      logFunction(payload);
+      const expectedLog: LumberjackLog<TestPayload> = {
+        message: testMessage,
+        level,
+        createdAt: fakeTime.getUnixEpochTicks(),
+        payload,
+        scope,
+      };
+
+      expect(((lumberjackService as unknown) as LumberjackService<TestPayload>).log).toHaveBeenCalledWith(expectedLog);
+    });
+
+    it('logs the specified static payload', () => {
+      const logFunction = builder.withPayload(payload).build();
+      logFunction();
+      const expectedLog: LumberjackLog<TestPayload> = {
+        message: testMessage,
+        level,
+        createdAt: fakeTime.getUnixEpochTicks(),
+        payload,
+        scope: '',
+      };
```

Comment:
```suggestion
      const expectedLog = createLog(level, testMessage, '', payload);
```

Add `createLog` to `@internal/test-util` imports.